Burst InSAR Product Guide (#360)
ASFHyP3/hyp3-docs: docs/guides/burst_insar_product_guide.md

(plus atmospheric delays and noise).

The DEM that is used for HyP3 InSAR processing is the
[2021 Release of the Copernicus GLO-30 Public DEM](https://spacedata.copernicus.eu/collections/copernicus-digital-elevation-model "Copernicus DEM" ){target=_blank}
dataset
[publicly available on AWS](https://registry.opendata.aws/copernicus-dem/ "https://registry.opendata.aws/copernicus-dem" ){target=_blank},
which provides global coverage at 30-m pixel spacing.
For more information about the 2021 updates, see the 'Releases' section of
[this article](https://spacedata.copernicus.eu/collections/copernicus-digital-elevation-model "Copernicus DEM" ){target=_blank}.

The portion of the DEM that covers the input bursts is downloaded.

#### Download Orbit Files and Calibration Auxiliary Data Files

The orbit files are downloaded from the
[Copernicus Sentinels POD Data Hub](https://scihub.copernicus.eu/gnss/#/home "Copernicus Sentinels POD Data Hub" ){target=_blank}.

The calibration auxiliary data files are downloaded from the
[Sentinel-1 Mission Performance Center](https://sar-mpc.eu/ "Sentinel-1 Mission Performance Center" ){target=_blank}.

> TODO: add more detail? i.e. what the files are and what they're used for

Reviewer: @hjkristenson Feel free to add some more detail regarding the orbit and auxcal files if you're interested, otherwise we could probably just leave it as-is for now.

Reviewer 2: I'll pass this responsibility along to @forrestfwilliams or @asjohnston-asf or anyone else who's actually worked with the orbit/calibration files.

{% extends "guides/insar_product_guide_template.md" %}

{% block header %}
# Sentinel-1 Burst InSAR Product Guide

This document is a guide for users of Sentinel-1 Burst Interferometric Synthetic Aperture Radar (InSAR) products generated by the Alaska Satellite Facility (ASF). 

### Sentinel-1 Bursts

> Data from the Sentinel-1 mission that is suitable for use in interferometry has historically been packaged into Single Look Complex (SLC) products. These SLCs include three subswaths, each containing many individual bursts. The framing of the SLCs is not consistent through time, so when using SLCs as the basis for InSAR, datasets do not always fully overlap. 

Author: "datasets do not fully overlap" doesn't paint the picture in my head, maybe "individual scenes" instead?
```suggestion
Data from the Sentinel-1 mission that is suitable for use in interferometry has historically been packaged into Single Look Complex (SLC) products. These SLCs include three subswaths, each containing many individual bursts. The framing of the SLCs is not consistent through time, so when using SLCs as the basis for InSAR, individual scenes do not always fully overlap. 
```

Reviewer 2: addressed in [this commit](https://github.com/ASFHyP3/hyp3-docs/pull/360/commits/1f1f0c1e82b857c14a8361b21b62b4f855ac2af9)

**1. Bursts are consistently geolocated through time**  
The coverage of a burst ID is the same for every orbit of the satellite, so you can be confident that every burst in a stack of acquisitions will cover the same geographic location. 

**2. Bursts cover a smaller geographic area**  
SLC products are extremely large, and in many cases, only a small portion of the image is of interest. You can process only the bursts that cover your specific area of interest, which significantly decreases the time and cost required to generate InSAR products.

Refer to the [Sentinel-1 Bursts tutorial](https://storymaps.arcgis.com/stories/88c8fe67933340779eddef212d76b8b8) to learn more about how ASF extracts burst-level products from Sentinel-1 SLCs.

### Burst InSAR Processing

The Sentinel-1 Burst InSAR products are generated using [ISCE2 software](https://github.com/isce-framework/isce2#readme "https://github.com/isce-framework/isce2" ){target=_blank}. ASF is committed to transparency in product development, and we are pleased to be able to offer an InSAR product that leverages open-source software for processing. 

> For those who would prefer to work at the scale of a full SLC, our original [On Demand InSAR](insar_product_guide.md) products are still available. These products have a larger footprint, and are generated using GAMMA software.     

Author: ```suggestion
For those who would prefer to work at the scale of a full SLC, our original [On Demand InSAR](insar_product_guide.md) products are still available. These products have a larger footprint, and are generated using [GAMMA software](https://www.gamma-rs.ch/software).
```

{% extends "guides/insar_product_guide_template.md" %}

{% block header %}
# Sentinel-1 Burst InSAR Product Guide

This document is a guide for users of Sentinel-1 Burst Interferometric Synthetic Aperture Radar (InSAR) products generated by the Alaska Satellite Facility (ASF). 

### Sentinel-1 Bursts

> Data from the Sentinel-1 mission that is suitable for use in interferometry has historically been packaged into Single Look Complex (SLC) products. These SLCs include three subswaths, each containing many individual bursts. The framing of the SLCs is not consistent through time, so when using SLCs as the basis for InSAR, scene pairs do not always fully overlap. 

Reviewer 3: ```suggestion
Single Look Complex (SLC) data from the Sentinel-1 mission that is suitable for use in interferometry has historically been packaged into Interferometric Wide (IW) SLC products. These IW SLC products include three subswaths, each containing many individual burst SLCs. The framing of the IW SLCs is not consistent through time, so when using IW SLCs as the basis for InSAR, scene pairs do not always fully overlap. 
```

SLC is a shorthand for the Sentinel-1 product format, but it is also the type of data contained within these products. I think it's an important to make the distinction between the products themselves and the data they contain so that it's clear that the burst products are also SLCs, just of a different size.

Reviewer 2: Do we still limit InSAR processing to just IW? Or can users submit EW SLCs for processing? Just making sure we don't need to broaden our references to IWs to *also* include EWs.

Author: We don't support EW. The HyP3 API requires "IW" be in the burst granule names. The hyp3-isce2 plugin failed to process the first two EW bursts I tried, one complaining "swath must be 1,2,3" and another complaining that no annotation .xml could be found in our mocked .SAFE directory.

Reviewer 3: Yep of the time being we only support IW.

> In contrast, working at the burst level of the Sentinel-1 data provides a couple key benefits: 

Reviewer 3: ```suggestion
In contrast, working at the burst level of the Sentinel-1 SLC data provides a couple key benefits: 
```

- The product type (always INT for InSAR) and the pixel spacing, which will be either 80, 40, or 20, based upon the number of looks selected when the job was submitted for processing
- The filename ends with the ASF product ID, a 4 digit hexadecimal number

![Figure 4](../images/asf_burst_insar_names.png "Breakdown of ASF InSAR Naming Scheme")

*Figure 4: Breakdown of ASF Burst InSAR naming scheme.*

### Image Files
# TODO: New Story Map
All of the main InSAR product files are 32-bit floating-point single-band GeoTIFFs. To learn more about the rasters included in the product package, refer to the [Exploring Sentinel-1 InSAR StoryMap](https://storymaps.arcgis.com/stories/8be186e4125741518118d0102e6835e5 "Exploring Sentinel-1 InSAR StoryMap https://arcg.is/11DaW90" ){target=_blank} tutorial.

- The *coherence* file pixel values range from 0.0 to 1.0, with 0.0 being completely non-coherent and 1.0 being perfectly coherent. 
- The *unwrapped phase* file shows the results of the phase unwrapping process. Negative values indicate movement towards the sensor, and positive values indicate movement away from the sensor. This is the main interferogram output.

> - The *wrapped phase* file indicates the interferogram phase after applying the adaptive filter immediately before unwrapping. Values range from negative pi to positive pi.

Author: ```suggestion
- The *wrapped phase* file indicates the interferogram phase after applying the adaptive filter immediately before unwrapping. Values range from negative pi to positive pi.
- The *connected components* file delineates regions unwrapped as contiguous units by the SNAPHU unwrapping algorithm. In general, products with fewer and larger regions will have higher quality phase unwrapping results than products with many small regions. This file is used by post-processors such as MintPy to correct unwrapping errors.
```

> - The *look vectors* theta (θ) and phi (φ) describe the elevation and orientation angles of the sensor's look direction.

Author: @cirrusasf I think here under "Image Files" is the best place to describe the look vectors in more detail

Reviewer 4: yes, I just edited "the look vectors" in Image Files section.

1. The **number of looks** drives the resolution and pixel spacing of the output products:

    | Looks | Resolution | Pixel Spacing |
    |-------|------------|---------------|
    | 20x4  | 160 m      | 80 m          |
    | 10x2  | 80 m       | 40 m          |
    | 5x1   | 40 m       | 20 m          |

    Products generated with 10x2 looks have a file size roughly 4 times that of 20x4-look products.
    Similarly, 5x1-look products have a file size roughly 4 times that of 10x2-look products
    (or 16 times that of 20x4-look products).

    The default is 20x4 looks.

2. There is an option to apply a **water mask** after phase unwrapping.
   This mask includes coastal waters and large inland waterbodies.
   A GeoTIFF of the water mask is always included with the InSAR product package,
   but when this option is selected, the water mask will be applied to
   the wrapped interferogram, the unwrapped interferogram, and the browse image.
   Water masking is turned off by default. Refer to the [Apply Water Mask section](#apply-water-mask) for more information about the water mask and how it is used.

{% endblock %}

{% block workflow %}
## InSAR Workflow

> TODO

Reviewer: ```suggestion
The Burst InSAR workflow used in HyP3 was developed by ASF using ISCE2 software. The steps include pre-processing, interferogram preparation, and product creation. Once these steps are performed, an output product package is created. See product packaging for details on the individual files included in the package.
```

> ## InSAR Workflow
> The Burst InSAR workflow used in HyP3 was developed by ASF using ISCE2 software. The steps include pre-processing, interferogram preparation, and product creation. Once these steps are performed, an output product package is created. See [product packaging](#product-packaging "Jump to Product Packaging section of this document") for details on the individual files included in the package.

Reviewer: ```suggestion
## Burst InSAR Workflow

The Burst InSAR workflow used in HyP3 was developed by ASF using ISCE2 software. The steps include pre-processing, interferogram preparation, and product creation. Once these steps are performed, an output product package is created. See [product packaging](#product-packaging "Jump to Product Packaging section of this document") for details on the individual files included in the package.
```

Users can optionally choose to apply the water mask to output products, which affects the wrapped interferogram,
the unwrapped interferogram, and the browse image. Areas covered by the water mask in these output images are set to
NoData.

Applying a water mask to an interferogram is only supported *after* phase unwrapping. Note that
applying the mask after phase unwrapping does not prevent unwrapping errors caused by the inclusion of water pixels
as valid data during the phase unwrapping process. When phase unwrapping occurs over large expanses of water, it can
lead to unexpected deformation signals or phase jumps in the unwrapped outputs, and the current masking approach
does not correct for these impacts.

#### Product Creation
Image files are exported into the widely-used GeoTIFF format in a Universal Transverse Mercator (UTM) projection. Images
are resampled to a pixel size that reflects the resolution of output image based on the requested number of looks: 80m
pixel size for 20x4 looks, 40m for 10x2 looks, and 20m for 5x1 looks.

Supporting metadata files are created, as well as a quick-look browse image.

{% endblock %}

{% block packaging %}
## Product Packaging

> HyP3 Burst InSAR output is a zip file containing various files, including GeoTIFFs, PNG browse images with geolocation information, Google Earth KMZ files, a metadata file, and a README file.

Author: ```suggestion
HyP3 Burst InSAR output is a zip file containing various files, including GeoTIFFs, a PNG browse image, a metadata file, and a README file.
```
We don't include a KMZ or multiple browse images for burst products like we do for full insar products.

### Naming Convention

The Burst InSAR product names are packed with information pertaining to the processing of the data, presented in the following order, as illustrated in Figure 4. 

> - The platform names, either Sentinel-1A or Sentinel-1B, are abbreviated "A" or "B", indicating the reference and secondary granule's imaging platform

Author: ```suggestion
- The imaging platform name, always S1 for Sentinel-1.
```
We don't indicate the specific satellite (A vs B) for burst insar products.

- Relative burst ID values assigned by ESA. Each value identifies a consistent burst footprint; relative burst ID values differ from one subswath to the next.

> - The reference start date and time and the secondary start dates, t.

Author: ```suggestion
- The acquisition dates of the reference (older) scene and the secondary (newer) scene
```

- The polarizations for the pair, either HH or VV.

> - The product type (always INT for InSAR) and the pixel spacing, which will be either 80 or 40, based upon the number of looks selected when the job was submitted for processing

Author: ```suggestion
- The product type (always INT for InSAR) and the pixel spacing, which will be either 80, 40, or 20, based upon the number of looks selected when the job was submitted for processing
```
We also offer 5x1 looks for burst insar, with a pixel size of 20m.

- The filename ends with the ASF product ID, a 4 digit hexadecimal number

![Figure 4](../images/asf_burst_insar_names.png "Breakdown of ASF InSAR Naming Scheme")

*Figure 4: Breakdown of ASF Burst InSAR naming scheme.*

### Image Files
# TODO: New Story Map
All of the main InSAR product files are 32-bit floating-point single-band GeoTIFFs. To learn more about the rasters included in the product package, refer to the [Exploring Sentinel-1 InSAR StoryMap](https://storymaps.arcgis.com/stories/8be186e4125741518118d0102e6835e5 "Exploring Sentinel-1 InSAR StoryMap https://arcg.is/11DaW90" ){target=_blank} tutorial.

- The *coherence* file pixel values range from 0.0 to 1.0, with 0.0 being completely non-coherent and 1.0 being perfectly coherent. 
- The *unwrapped phase* file shows the results of the phase unwrapping process. Negative values indicate movement towards the sensor, and positive values indicate movement away from the sensor. This is the main interferogram output.

> - The *wrapped phase* file indicates the interferogram phase after applying the adaptive filter immediately before unwrapping. Values range from negative pi to positive pi. *(optional)*
> - The *look vectors* theta (θ) and phi (φ) describe the elevation and orientation angles of the sensor's look direction. *(optional)*
> - The *DEM* file gives the local terrain heights in meters, with a geoid correction applied. *(optional)*

Author: None of the output files are optional for bursts, we always include everything at the moment.
```suggestion
- The *wrapped phase* file indicates the interferogram phase after applying the adaptive filter immediately before unwrapping. Values range from negative pi to positive pi.
- The *look vectors* theta (θ) and phi (φ) describe the elevation and orientation angles of the sensor's look direction.
- The *DEM* file gives the local terrain heights in meters, with a geoid correction applied.
```

Reviewer 4: There is no place to describe the range of the lv_theta and lv_phi. I think we should add the valid ranges for both lv_theta and lv_phi. I will do it.

- The *water mask* file indicates coastal waters and large inland waterbodies. Pixel values of 1 indicate land and 0 indicate water. This file is in 8-bit unsigned integer format.

If the **water mask** option is selected, the water mask is applied *after* phase unwrapping to exclude water pixels from the output. The water mask is generated using the [GSHHG](http://www.soest.hawaii.edu/wessel/gshhg "http://www.soest.hawaii.edu/wessel/gshhg/land" ){target=_blank} dataset. To compile the reference shapefile, the full-resolution L1 dataset (boundary between land and ocean) and L5 dataset (boundary between Antarctic ice and ocean) were combined. The L3 dataset (boundary between islands and the lakes they are within) was removed from the L2 dataset (boundary between lakes and land), and this combined dataset was removed from the combined L1/L5 dataset. The GSHHG dataset was last updated in 2017, so there may be discrepancies where shorelines have changed.

> A **browse image** is included for the unwrapped (unw_phase) phase file, which is in PNG format and is 2048 pixels wide. The browse image is displayed using a cyclic color ramp to generate fringes. Each fringe in the browse image represents a phase difference of 6 pi. Because each 2-pi difference is equivalent to half the wavelength of the sensor, each 6-pi fringe represents about 8.3 cm of line-of-sight displacement for these Sentinel-1 products.

Author: ```suggestion
A **browse image** is included for the unwrapped (unw_phase) phase file, which is in PNG format and is 2048 pixels wide.
```
Our browse images for burst products are a simple greyscale, we don't use a color ramp like we do for full insar products.
![S1_136231_IW2_20200604_20200616_VV_INT80_D94E_unw_phase](https://github.com/ASFHyP3/hyp3-docs/assets/17994518/3ba2c546-ad6f-4456-899d-c72237156e08)

*Table 3: Metadata files in product package*

#### README File
The text file with extension .README.md.txt explains the files included in the folder, and is customized to reflect that particular product. Users unfamiliar with InSAR products should start by reading this README file, which will give some background on each of the files included in the product folder.

#### InSAR Parameter File
The text file with extension .txt includes processing parameters used to generate the InSAR product as well as metadata attributes for the InSAR pair.  These are detailed in Table 4.  

| Name                                                      | Description                                                                                                 | Possible Value                                                                                              |
|-----------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|

> | Reference Granule                                         | ESA granule name for reference scene (of the two scenes in the pair, the dataset with the oldest timestamp) | S1A<wbr>_IW<wbr>_SLC<wbr>__1SDV<wbr>_20200116T032559<wbr>_20200116T032627<wbr>_030820<wbr>_038928<wbr>_F5DC |
> | Secondary Granule                                         | ESA granule name for secondary scene (of the two scenes in the pair, the dataset with the newest timestamp) | S1B<wbr>_IW<wbr>_SLC<wbr>__1SDV<wbr>_20200128T032559<wbr>_20200128T032627<wbr>_030995<wbr>_038F51<wbr>_7D4F |

Author: ```suggestion
| Reference Granule                                         | Granule name for reference burst (of the two scenes in the pair, the dataset with the oldest timestamp) | S1<wbr>_136231<wbr>_IW2<wbr>_20200604T022312<wbr>_VV<wbr>_7C85-BURST |
| Secondary Granule                                         | Granule name for secondary burst (of the two scenes in the pair, the dataset with the newest timestamp) | S1<wbr>_136231<wbr>_IW2<wbr>_20200616T022313<wbr>_VV<wbr>_5D11-BURST |
```

| Slant range center                                        | Distance in meters from satellite to the center point imaged                                                | 879794.1404                                                                                                 | 
| Slant range far                                           | Distance in meters from satellite to farthest point imaged                                                  | 960070.8469                                                                                                 |
| Range looks                                               | Number of looks taken in the range direction                                                                | 20                                                                                                          | 
| Azimuth looks                                             | Number of looks taken in the azimuth direction                                                              | 4                                                                                                           |

> | InSAR phase filter                                        | Name of the phase filter used                                                                               | adf                                                                                                         | 
> | Phase filter parameter                                    | Dampening factor                                                                                            | 0.6                                                                                                         |
> | Resolution of output (m)                                  | Pixel spacing in meters for output products                                                                 | 80                                                                                                          | 
> | Range bandpass filter                                     | Range bandpass filter applied                                                                               | no                                                                                                          |
> | Azimuth bandpass filter                                   | Azimuth bandpass filter applied                                                                             | no                                                                                                          |
> | DEM source                                                | DEM used in processing                                                                                      | GLO-30                                                                                                      |
> | DEM resolution                                            | Pixel spacing in meters for DEM used to process this scene                                                  | 160                                                                                                         |
> | Unwrapping type                                           | Phase unwrapping algorithm used                                                                             | mcf                                                                                                         |
> | Phase at Reference Point                                  | Original unwrapped phase value at the reference point (set to 0 in output unwrapped phase raster)           | -4.21967                                                                                                    |
> | Azimuth line of the reference point in SAR space          | Row number (in SAR space) of the reference point                                                            | 2737.0                                                                                                      |
> | Range pixel of the reference point in SAR space           | Column number (in SAR space) of the reference point                                                         | 739.0                                                                                                       |
> | Y coordinate of the reference point in the map projection | Latitude of the reference point in projected coordinates (UTM Zone - meters)                                | 4112453.3223                                                                                                |
> | X coordinate of the reference point in the map projection | Longitude of the reference point in projected coordinates (UTM Zone - meters)                               | 589307.6248                                                                                                 |
> | Latitude of the reference point (WGS84)                   | Latitude of the reference point in WGS84 Geographic Coordinate System (degrees)                             | 37.1542125                                                                                                  |
> | Longitude of the reference point (WGS84)                  | Longitude of the reference point in WGS84 Geographic Coordinate System (degrees)                            | 40.00574707                                                                                                 |
> | Unwrapping threshold                                      | Minimum coherence required to unwrap a given pixel                                                          | none                                                                                                        |
> | Speckle filter                                            | Speckle filter applied                                                                                      | no                                                                                                          |

Author: Updating to match example output
[S1_136231_IW2_20200604_20200616_VV_INT80_D94E.txt](https://github.com/ASFHyP3/hyp3-docs/files/12693515/S1_136231_IW2_20200604_20200616_VV_INT80_D94E.txt)

```suggestion
| InSAR phase filter                                        | Name of the phase filter used                                                                               | yes                                                                                                         | 
| Phase filter parameter                                    | Dampening factor                                                                                            | 0.5                                                                                                         |
| Range bandpass filter                                     | Range bandpass filter applied                                                                               | no                                                                                                          |
| Azimuth bandpass filter                                   | Azimuth bandpass filter applied                                                                             | no                                                                                                          |
| DEM source                                                | DEM used in processing                                                                                      | GLO-30                                                                                                      |
| DEM resolution                                            | Pixel spacing in meters for DEM used to process this scene                                                  | 30                                                                                                          |
| Unwrapping type                                           | Phase unwrapping algorithm used                                                                             | snaphu_mcf                                                                                                  |
| Speckle filter                                            | Speckle filter applied                                                                                      | yes                                                                                                         |
```

A single interferogram cannot be used to determine the relative contributions of vertical and horizontal movement to the line-of-sight displacement measurement. To determine how much of the signal is driven by vertical vs. horizontal movement, you must either use a time series of interferograms, or use reference measurements with known vertical and horizontal components (such as GNSS measurements from the region of deformation) to deconstruct the line-of-sight displacement.

{% endblock %}

{% block reference_point %}
### Phase Unwrapping Reference Point
The reference point for phase unwrapping is set automatically by the topsApp.py script. It may not be an ideal location to use as a reference point for phase unwrapping. If it is located in an area undergoing deformation, or in an area with low coherence, the unwrapping may be of lower quality than if the reference point was in a more suitable location.

> Even when there are not phase unwrapping errors introduced by phase discontinuities, it is important to be aware that unwrapped phase differences are calculated relative to the reference point.[(TODO: check if this is the same with ISCE2:) The phase difference value of the reference point is set to 0 during phase unwrapping, so any displacement values will be relative to that benchmark.] 

Reviewer 2: Verify what we know (or don't) about how topsApp selects a reference point. Maybe @forrestfwilliams has some insight?

Reviewer 3: This section looks accurate to me. I don't have any more information on how topsApp picks the reference, but I don't think we need to add any more information.

Reviewer 2: OK. I've cleaned up the TODO, so these comments should be reflected in [this commit](https://github.com/ASFHyP3/hyp3-docs/pull/360/commits/122437f86a3c7268fd037e07e7c964408797ed4a)

If you are interested in the amount of displacement in a particular area, you may wish to choose your own reference point. The ideal reference point would be in an area of high coherence beyond where deformation has occurred. The unwrapped phase measurements can be adjusted to be relative to this new reference point. To adjust the values in the unwrapped phase GeoTIFF, simply select a reference point that is optimal for your use case and subtract the unwrapped phase value of that reference point from each pixel in the unwrapped phase raster:

**ΔΨ<sup>&ast;</sup>** = **ΔΨ** - Δψ<sub>ref</sub>

where **ΔΨ<sup>&ast;</sup>** is the adjusted unwrapped phase, **ΔΨ** is the original unwrapped phase, and Δψ<sub>ref</sub> is the unwrapped phase value at the new reference point.

#### Displacement Values from a Single Interferogram

> In general, calculating displacement values from a single interferogram is not recommended. It will be more robust to use a time series approach to more accurately determine the pattern of movement. When using SAR time-series software such as [MintPy](https://mintpy.readthedocs.io/en/latest/ "https://mintpy.readthedocs.io/en/latest" ){target=_blank}, you have the option to select a specific reference point, and the values of the input rasters will be adjusted accordingly.

Reviewer 2: Are the burst products supported in MintPy?

Reviewer: I believe the answer is yes, and we should be releasing a tutorial as part of https://asfdaac.atlassian.net/browse/TOOL-2176.

Reviewer 3: yes, they are supported.

- topsapp.swap_burst_vrts(): Switch the reference and secondary bursts to use the burst data download from ASF
- topsapp.run_topsapp_burst(start='computeBaselines', end='unwrap2stage', config_xml=config_path):
    Run the remaining processing steps including:
    - Calculate the perpendicular and parallel baselines
    - Verify the DEM file to make sure it covers the bursts
    - Map DEM into the radar coordinates of the reference image. This generates the longitude,
            latitude, height and LOS angles on a pixel by pixel grid for each burst.
    - Estimate the azimuth offsets between the input SLC bursts (The Enhanced Spectral Diversity (ESD) method is NOT used)
    - Estimate the range offsets between the input SLC bursts
    - Coregister the secondary SLC burst by applying the estimated range and azimuth offsets
    - Produce the wrapped phase interferogram
    - Unwrap the wrapped phase interferogram using SNAPHU to produce the wrapped phase interferogram
- topsapp.run_topsapp_burst(start='geocode', end='geocode', config_xml=config_path): Geocode the output products

### Post-Processing

#### Apply Water Mask

> A water mask identifying coastal waters and major inland waterbodies is generated using the Global Self-consistent,
> Hierarchical, High-resolution Geography Database (GSHHG) dataset (https://www.ngdc.noaa.gov/mgg/shorelines). This water mask raster is always included with the Burst InSAR products for reference, but is not applied to the interferometry products by default.

Reviewer: ```suggestion
A water mask identifying coastal waters and major inland waterbodies is generated using the [Global Self-consistent,
Hierarchical, High-resolution Geography Database (GSHHG)](http://www.soest.hawaii.edu/wessel/gshhg "http://www.soest.hawaii.edu/wessel/gshhg" ){target=_blank} dataset. This water mask raster is always included with the Burst InSAR products for reference, but is not applied to the interferometry products by default.
```

- The *look vectors* theta (θ) and phi (φ) describe the elevation and orientation angles of the look vector in radians. The look vectors refer to the look direction back towards the sensor. 
    - The *lv_theta* (θ) file indicates the SAR look vector elevation angle at each pixel, ranging from -π/2 (down) to π/2 (up). The look vector elevation angle is defined as the angle between the horizontal surface and the look vector with positive angles indicating sensor positions above the surface. 
    - The *lv_phi* (φ) file indicates the SAR look vector orientation angle at each pixel. The look vector orientation angle is defined as the angle between the East direction and the projection of the look vector on the horizontal surface plane. The orientation angle increases towards north, with the North direction corresponding to π/2 (and south to -π/2). The orientation angle range is -π to π.
- The *DEM* file gives the local terrain heights in meters, with a geoid correction applied.
- The *water mask* file indicates coastal waters and large inland waterbodies. Pixel values of 1 indicate land and 0 indicate water. This file is in 8-bit unsigned integer format.

> If the **water mask** option is selected, the water mask is applied *after* phase unwrapping to exclude water pixels from the output. The water mask is generated using the [GSHHG](http://www.soest.hawaii.edu/wessel/gshhg "http://www.soest.hawaii.edu/wessel/gshhg/land" ){target=_blank} dataset. To compile the reference shapefile, the full-resolution L1 dataset (boundary between land and ocean) and L5 dataset (boundary between Antarctic ice and ocean) were combined. The L3 dataset (boundary between islands and the lakes they are within) was removed from the L2 dataset (boundary between lakes and land), and this combined dataset was removed from the combined L1/L5 dataset. The GSHHG dataset was last updated in 2017, so there may be discrepancies where shorelines have changed.

Reviewer: The URL ending in `/land` returns 404.
```suggestion
If the **water mask** option is selected, the water mask is applied *after* phase unwrapping to exclude water pixels from the output. The water mask is generated using the [GSHHG](http://www.soest.hawaii.edu/wessel/gshhg "http://www.soest.hawaii.edu/wessel/gshhg" ){target=_blank} dataset. To compile the reference shapefile, the full-resolution L1 dataset (boundary between land and ocean) and L5 dataset (boundary between Antarctic ice and ocean) were combined. The L3 dataset (boundary between islands and the lakes they are within) was removed from the L2 dataset (boundary between lakes and land), and this combined dataset was removed from the combined L1/L5 dataset. The GSHHG dataset was last updated in 2017, so there may be discrepancies where shorelines have changed.
```

{% extends "guides/insar_product_guide_template.md" %}

{% block header %}
# Sentinel-1 Burst InSAR Product Guide

TODO
{% endblock %}

{% block workflow %}
## InSAR Workflow
TODO

### Pre-Processing

#### Download SLC Data
TODO

#### Prepare the DEM File
TODO

#### Download Orbit and Auxiliary Data Files

> TODO

Author: @jtherrmann is drafting Pre-Processing

### InSAR Processing

> TODO

Author: @cirrusasf is drafting the details of the ISCE processing workflow

{% extends "guides/insar_product_guide_template.md" %}

{% block header %}
# Sentinel-1 Burst InSAR Product Guide

TODO

> {% endblock %}

Reviewer 2: @hjkristenson will work on this section
